### PR TITLE
FEAT: Advanced Fields calculator

### LINKED ISSUE #4740: Create an extension to make Fields Calculator easier to use
### Description of the feature

There are some common operations with the Fields Calculator, that can be automated

### Steps for implementing the feature

_No response_

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
Close #4740

### CODE REVIEW

File: pyaedt/modules/fields_calculator.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    def add_expression(self, calculation, assignment, name=None):
+        """Add named expression.
+
+        Parameters
+        ----------
+        calculation :
+            Calculation type.
+        assignment : int or :class:`pyaedt.modeler.cad.object3d.Object3d` or
+         :class:`pyaedt.modeler.cad.FacePrimitive
+            Name of the object from which to add the named expression.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:60]:
```suggestion
            Name of the object to add the named expression from.
```
We use less formal writing here as this is how we speak and is easier to comprehend.

File: pyaedt/modules/fields_calculator.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        # Import clc
+        self.ofieldsreporter.LoadNamedExpressions(
+            os.path.abspath(file_name), expression_info["fields_type"][design_type_index], [name]
+        )
+
+        return expression_info["name"]
+
+    @pyaedt_function_handler()
+    def create_expression_file(self, name, operations):
+        """Create calculator expression file.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:139]:
```suggestion
        """Create a calculator expression file.
```

File: pyaedt/modules/fields_calculator.py [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+                for operation in operations:
+                    file.write(f"	{operation}\n")
+                file.write("$end 'Named_Expression'\n")
+        except Exception:  # pragma: no cover
+            return False
+        return os.path.abspath(file_name)
+
+    @pyaedt_function_handler()
+    def expression_plot(self, calculation, assignment, names, setup=None):
+        """Add named expression.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:172]:
```suggestion
        """Add a named expression.
```

File: pyaedt/modules/fields_calculator.py [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+        Parameters
+        ----------
+        calculation : str
+            Calculation type.
+        assignment : list
+            List of assignments to apply the expression. If the expression is not general, assignment is not needed.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:179]:
```suggestion
            List of assignments to apply the expression to. If the expression is not general, assignment is not needed.
```

File: pyaedt/modules/fields_calculator.py [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+                        if assign not in self.__app.modeler.point_names and assign not in self.__app.modeler.line_names:
+                            report = self.__app.post.create_fieldplot_volume(
+                                quantity=names[0], assignment=assign, field_type=fields_type, intrinsics=intrinsic
+                            )
+                            reports.append(report)
+        return reports
+
+    @pyaedt_function_handler()
+    def delete_expression(self, name=None):
+        """Delete named expression.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:256]:
```suggestion
        """Delete a named expression.
```

File: pyaedt/modules/fields_calculator.py [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the named expression. The default is ``None`` in which case all named expressions are deleted.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:261]:
```suggestion
            Name of the named expression. The default is ``None``, in which case all named expressions are deleted.
```

File: pyaedt/modules/fields_calculator.py [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+        if not name:
+            self.ofieldsreporter.ClearAllNamedExpr()
+            return True
+        if self.is_expression_defined(name):
+            self.ofieldsreporter.DeleteNamedExpr(name)
+        return True
+
+    @pyaedt_function_handler()
+    def is_expression_defined(self, name):
+        """Check if expression exists.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:277]:
```suggestion
        """Check if an expression exists.
```

File: pyaedt/modules/fields_calculator.py [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+            ``True`` when exists.
+        """
+        is_defined = self.ofieldsreporter.DoesNamedExpressionExists(name)
+        if is_defined == 1:
+            return True
+        return False
+
+    @pyaedt_function_handler()
+    def is_general_expression(self, name):
+        """Check if expression is general.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:296]:
```suggestion
        """Check if a named expression is general.
```

File: pyaedt/modules/fields_calculator.py [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+
+        Parameters
+        ----------
+        name : str,
+            Named expression.
+
+        Returns
+        -------
+        bool
+            ``True`` when exists.
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:306]:
```suggestion
            ``True`` if the named expression is general.
```

File: pyaedt/modules/fields_calculator.py [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+        new_expression_catalog = read_toml(input_file)
+
+        if new_expression_catalog:
+            self.expression_catalog.update(new_expression_catalog)
+
+        return self.expression_catalog
+
+    @staticmethod
+    def __has_integer(lst):  # pragma: no cover
+        """Check if list has integers."""
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:345]:
```suggestion
        """Check if a list has integers."""
```

File: pyaedt/modules/fields_calculator.py [hunk 11]
```diff
@@ -0,0 +1,356 @@
...
+        for item in lst:
+            if isinstance(item, int):
+                return True
+        return False
+
+    def __has_lines(self, lst):  # pragma: no cover
+        """Check if list has lines."""
```
Comment (PipKat) [inline pyaedt/modules/fields_calculator.py:352]:
```suggestion
        """Check if a list has lines."""
```

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Pull request merged**